### PR TITLE
feat(stonedb): Header file format standardization (#36)

### LINKED ISSUE #36: feature: Header file format standardization
**Is your feature request related to a problem? Please describe.**

~~~
#include "core/array_map.h"
#include "core/reactor.h"
#include "fmt/time.h"
#include "log_cli.h"
~~~

>*The header file format is standardized. For the above header files, some contain the directory and some have no path. The included paths should be standardized. Or write the include directory of the header file in cmake. Then, use the method of "path /header.h". The second method is recommended. Its advantage is that you can see its directory.*

**Describe the solution you'd like**

Bad:
~~~
#include "core/array_map.h"
#include "core/reactor.h"
#include "fmt/time.h"
#include "log_cli.h"
~~~
Valid:
~~~
#include "core/array_map.h"
#include "core/reactor.h"
#include "fmt/time.h"
#include "util/log_cli.h"
~~~

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by konghaiya**
## Summary about this PR

Issue Number: close #36

## Tests Check List

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [X] New Feature
- [ ] Bug Fix
- [X] Improvement
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

**Comment by RingsC** [comment]
@hustjieke , PTAL.

File: sql/sql_parse.cc [hunk 1]
```diff
@@ -3542,15 +3542,15 @@ case SQLCOM_PREPARE:
...
-          int bh_res, free_join_from_bh, optimize_after_bh;
+          int sdb_res, free_join_from_sdb, optimize_after_sdb;
 
           //FIXME: this doesn't work but just compile 
           select_result *sel_result2 = sel_result;
           if (stonedb::dbhandler::SDB_HandleSelect(thd, lex, sel_result2, (ulong)OPTION_SETUP_TABLES_DONE, 
-                              bh_res, optimize_after_bh, free_join_from_bh, (int)true)==0)
-            res= handle_select(thd, sel_result, OPTION_SETUP_TABLES_DONE, optimize_after_bh, free_join_from_bh);
+                              sdb_res, optimize_after_sdb, free_join_from_sdb, (int)true)==0)
+            res= handle_select(thd, sel_result, OPTION_SETUP_TABLES_DONE, optimize_after_sdb, free_join_from_sdb);
```
Comment (hustjieke) [inline sql/sql_parse.cc:3551]: Format code with `clang-format`

Comment (konghaiya) [reply sql/sql_parse.cc:3551]: This is the code of MySQL. No formatting is required.

Comment (hustjieke) [reply sql/sql_parse.cc:3551]: make sense!

File: storage/stonedb/base/core/future.h [hunk 2]
```diff
@@ -30,12 +30,12 @@
 #include <stdexcept>
 #include <type_traits>
 
-#include "../util/gcc6-concepts.h"
```
Comment (hustjieke) [inline storage/stonedb/base/core/future.h:33]: `#include "base/util/gcc6-concepts.h"` should be after `#include ""base/core/thread_impl.h"` by ordering alphabetically.

File: storage/stonedb/base/core/iostream.h [hunk 3]
```diff
@@ -34,9 +34,9 @@
 #define STONEDB_BASE_IOSTREAM_H_
 #pragma once
 
-#include "future.h"
-#include "scattered_message.h"
-#include "temporary_buffer.h"
+#include "base/core/future.h"
+#include "base/core/scattered_message.h"
```
Comment (hustjieke) [inline storage/stonedb/base/core/iostream.h:38]: `#include "base/core/scattered_message.h"` ordered alphabetically.

File: storage/stonedb/base/core/posix.h [hunk 4]
```diff
@@ -44,8 +44,8 @@
 #include <utility>
 
 #include <boost/optional.hpp>
-#include "net/socket_defs.h"
-#include "sstring.h"
+#include "base/net/socket_defs.h"
+#include "base/core/sstring.h"
```
Comment (hustjieke) [inline storage/stonedb/base/core/posix.h:47]: `#include "base/core/sstring.h"` should be ordered alphabetically.

File: storage/stonedb/base/core/reactor.cpp [hunk 5]
```diff
@@ -47,22 +47,22 @@
...
-#include "thread.h"
-#include "util/conversions.h"
-#include "util/log.h"
+#include "base/core/future_util.h"
+#include "base/core/posix.h"
+#include "base/core/file_impl.h"
+#include "base/net/packet.h"
+#include "base/net/posix_stack.h"
+#include "base/net/stack.h"
+#include "base/core/prefetch.h"
```
Comment (hustjieke) [inline storage/stonedb/base/core/reactor.cpp:53]: Should be ordered alphabetically here

File: storage/stonedb/base/core/reactor.cpp [hunk 6]
```diff
@@ -74,17 +74,17 @@
...
 #if defined(__x86_64__) || defined(__i386__)
 #include <xmmintrin.h>
 #endif
 
-#include "core/metrics.h"
-#include "exception_hacks.h"
-#include "execution_stage.h"
-#include "util/defer.h"
+#include "base/core/metrics.h"
+#include "base/core/exception_hacks.h"
```
Comment (hustjieke) [inline storage/stonedb/base/core/reactor.cpp:84]: Should be ordered alphabetically.

File: storage/stonedb/base/core/reactor.h [hunk 7]
```diff
@@ -61,27 +56,34 @@
...
-#include "temporary_buffer.h"
-#include "timer.h"
-#include "util/log.h"
+#include "base/core/deleter.h"
+#include "base/core/fair_queue.h"
+#include "base/core/file.h"
+#include "base/core/future.h"
+#include "base/core/lowres_clock.h"
+#include "base/core/manual_clock.h"
+#include "base/net/api.h"
```
Comment (hustjieke) [inline storage/stonedb/base/core/reactor.h]: Should be ordered alphabetically.

File: storage/stonedb/base/core/scattered_message.h [hunk 8]
```diff
@@ -27,10 +27,10 @@
 #include <memory>
 #include <vector>
 
-#include "core/deleter.h"
-#include "core/temporary_buffer.h"
-#include "net/packet.h"
-#include "sstring.h"
+#include "base/core/deleter.h"
+#include "base/core/temporary_buffer.h"
+#include "base/net/packet.h"
```
Comment (hustjieke) [inline storage/stonedb/base/core/scattered_message.h:33]: Should be ordered alphabetically.

File: storage/stonedb/base/core/scollectd.h [hunk 9]
```diff
@@ -34,14 +34,14 @@
...
-#include "core/sstring.h"
-#include "future.h"
-#include "net/byteorder.h"
-#include "util/log.h"
-
-#include "core/metrics_api.h"
+#include "base/core/print.h"
+#include "base/core/shared_ptr.h"
+#include "base/core/sstring.h"
+#include "base/core/future.h"
```
Comment (hustjieke) [inline storage/stonedb/base/core/scollectd.h]: Should be ordered alphabetically.

File: storage/stonedb/base/net/api.h [hunk 10]
```diff
@@ -29,13 +29,13 @@
...
 
-#include "core/future.h"
-#include "core/iostream.h"
-#include "core/print.h"
-#include "core/temporary_buffer.h"
-#include "net/byteorder.h"
-#include "net/packet.h"
-#include "net/socket_defs.h"
+#include "base/core/future.h"
+#include "base/core/iostream.h"
```
Comment (hustjieke) [inline storage/stonedb/base/net/api.h:33]: Should be ordered alphabetically.

File: storage/stonedb/base/net/packet.h [hunk 11]
```diff
@@ -30,9 +30,9 @@
 #include <iosfwd>
 #include <vector>
 
-#include "const.h"
-#include "core/deleter.h"
-#include "core/temporary_buffer.h"
+#include "base/net/const.h"
```
Comment (hustjieke) [inline storage/stonedb/base/net/packet.h]: Should be ordered alphabetically.

File: storage/stonedb/base/util/log_cli.h [hunk 12]
```diff
@@ -23,14 +23,13 @@
...
-#include "log.h"
-
 #include <algorithm>
 #include <iostream>
 #include <unordered_map>
 
-#include "core/sstring.h"
-#include "program_options.h"
+#include "base/util/log.h"
+#include "base/core/sstring.h"
```
Comment (hustjieke) [inline storage/stonedb/base/util/log_cli.h:30]: Should be ordered alphabetically.

File: storage/stonedb/common/mysql_gate.h [hunk 13]
```diff
@@ -36,22 +35,21 @@
...
-#include "mysql_com.h"
 #include "mysqld_suffix.h"
 #include "rpl_slave.h"
 #include "sql_optimizer.h"
-
 #include "key.h"
+#include "lock.h"
+#include "transaction.h"
+
+#include "probes_mysql.h"
```
Comment (hustjieke) [inline storage/stonedb/common/mysql_gate.h]: Should be ordered alphabetically.

File: storage/stonedb/compress/num_compressor.h [hunk 14]
```diff
@@ -18,15 +18,15 @@
...
 
-#include "basic_data_filt.h"
+#include "compress/basic_data_filt.h"
 #include "common/assert.h"
 #include "core/quick_math.h"
 #include "core/tools.h"
-#include "defs.h"
-#include "part_dict.h"
-#include "range_code.h"
+#include "compress/defs.h"
```
Comment (hustjieke) [inline storage/stonedb/compress/num_compressor.h:23]: Should be ordered alphabetically.

**Comment by konghaiya** [comment]
The StoneDB code has been formatted

File: storage/stonedb/index/rc_table_index.cpp [hunk 15]
```diff
@@ -27,7 +27,7 @@
 namespace stonedb {
 namespace index {
 
-using ROCKSDB_NAMESPACE::Status;
```
Comment (hustjieke) [inline storage/stonedb/index/rc_table_index.cpp]: Why deleted `ROCKSDB_NAMESPACE` here?

**Comment by hustjieke** [comment]
Fix the commit style: 
`feature: Header file format standardization #36`
-->
`feat(stonedb) : ..... (#36)`

File: storage/stonedb/index/rc_table_index.cpp (hunk 15 above)
Comment (konghaiya) [reply storage/stonedb/index/rc_table_index.cpp]: Corrected.

**Review by hustjieke — APPROVED** [review]

**Comment by hustjieke** [comment]
LGTM
PTAL CR @RingsC

File: storage/stonedb/core/pack_str.cpp [hunk 16]
```diff
@@ -27,14 +27,13 @@
 #include "compress/text_compressor.h"
 #include "core/bin_tools.h"
 #include "core/column_share.h"
-#include "core/pack_str.h"
```
Comment (RingsC) [inline storage/stonedb/core/pack_str.cpp:30]: Does 'core/pack_str.h' need anymore ?

Comment (konghaiya) [reply storage/stonedb/core/pack_str.cpp:30]: This '#include' has been moved to the top

**Comment by RingsC** [comment]
LGTM

**Pull request merged**